google/cloud/spanner/integration_tests/client_integration_test.cc

```diff
@@ -814,6 +814,33 @@ TEST_F(ClientIntegrationTest, ExecuteQueryExactStalenessDuration) {
   });
 }
 
+/// @test Test that a directed read within a read-write transaction fails.
+TEST_F(ClientIntegrationTest, DirectedReadWithinReadWriteTransaction) {
+  auto& client = *client_;
+  auto commit =
+      client_->Commit([&client](Transaction const& txn) -> StatusOr<Mutations> {
+        auto rows =
+            client.Read(txn, "Singers", KeySet::All(),
+                        {"SingerId", "FirstName", "LastName"},
+                        Options{}.set<DirectedReadOption>(IncludeReplicas(
+                            {ReplicaSelection(ReplicaType::kReadOnly)},
+                            /*auto_failover_disabled=*/true)));
+        using RowType = std::tuple<std::int64_t, std::string, std::string>;
+        for (auto& row : StreamOf<RowType>(rows)) {
+          if (!row) return row.status();
+        }
+        return Mutations{};
+      });
+  if (UsingEmulator()) {
+    EXPECT_THAT(commit, IsOk());  // The emulator doesn't diagnose the error.
```

Comment:
Maybe we should simply put a `if (UsingEmulator()) GTEST_SKIP();` at the beginning of the test.

Reply:
I prefer these slightly more complicated incantations because they will actively alert us when the behavior of the emulator changes (rather than just ignoring it [most probably] forevermore).